### ISSUE TITLE
`#with_role` + `merge` = 🙅‍♂️

### ISSUE THREAD

**Issue by KevinColemanInc** [RolifyCommunity/rolify#521]
Lets say I have models: `course`, `course_questions`.  A user can be an `editor` of a course. being an editor of a course means you can modify related models, like `course_questions`

Lets say I want to find all of the `course_questions` a user can modify. I should be able to do that with a query like:

```
managed_courses = Course.with_role(:editor, user)
CourseQuestion.joins(:course).merge(managed_courses)
```

But, b/c `with_role` is using a `select`, it messes up the ids! So instead of selecting the attributes of `course_question`, it actually selects the attributes of `course` and this swapping out the id.

If I do
```
course_questions = CourseQuestion.joins(:course).merge(managed_courses)
course_questions.first.id # This is actually the id of the course, not the id course_modules.
```

TL;DR: Don't use the active record `#merge` method to determine if a user's role works on sub-resources.

My quick fix is:
```
managed_courses = Course.with_role(:editor, user)
CourseQuestion.joins(:course).where(courses: {id: managed_courses.pluck(:id)})
```
